sql-plugin/src/main/scala/com/nvidia/spark/rapids/AbstractGpuJoinIterator.scala

```diff
@@ -200,53 +203,133 @@ abstract class SplittableJoinIterator(
     val wasInitialJoin = isInitialJoin
     isInitialJoin = false
     if (pendingSplits.nonEmpty || stream.hasNext) {
-      val cb = if (pendingSplits.nonEmpty) {
-        opTime.ns {
-          withResource(pendingSplits.dequeue()) {
-            _.getColumnarBatch()
-          }
-        }
+      val scb = if (pendingSplits.nonEmpty) {
+        pendingSplits.dequeue()
       } else {
         val batch = withResource(stream.next()) { lazyBatch =>
           opTime.ns {
-            lazyBatch.releaseBatch()
+            SpillableColumnarBatch(lazyBatch.releaseBatch(),
+              SpillPriorities.ACTIVE_ON_DECK_PRIORITY, spillCallback)
           }
         }
         batch
       }
       opTime.ns {
-        withResource(cb) { cb =>
-          val numJoinRows = computeNumJoinRows(cb)
+        withResource(scb) { _ =>
+          val numJoinRows = computeNumJoinRows(scb)
 
           // We want the gather maps size to be around the target size. There are two gather maps
           // that are made up of ints, so compute how many rows on the stream side will produce the
           // desired gather maps size.
           val maxJoinRows = Math.max(1, targetSize / (2 * Integer.BYTES))
```

Comment:
For now it's probably OK to keep this out of the retry loops, but I think we should figure out if we need to split on a retry.